### PR TITLE
fix: when setting isOidcEnabled is false override env variable

### LINKED ISSUE #3619: fix: when setting isOidcEnabled is false, I can still log in via oidc
Observed by Brenda and Dieuwke.

Expected behavior:
- when I disable isOidcEnabled then both UI and backend are disabled.
- the OIDC client etc settings is via environment

**What version of EMX2 are you using (see footer)**
v10.48.8

**Describe the bug**
When setting isOidcEnabled is false, I can still log in via oidc

**To Reproduce**
1. Go to https://data-catalogue.molgeniscloud.org/apps/central/#/admin/settings and log in as admin
2.  See that setting isOidcEnabled is false, or set isOidcEnabled to false if it is true
3. Log out.
4. Log in via oidc.
5. See that you can still log in via oidc.

On the directory app this stays on/off. Check if there the menu doesn't look at the setting?

**Expected behavior**

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by harmbrugge**
Fixes #3619

What are the main changes you did:
- When the database setting isOidcEnabled is set to false it will used this setting and overrides env OIDC properties

how to test:
- Spin up an instance of emx2 with OIDC env variables set
- See OIDC is enabled 
- Log in with admin
- Set isOidcEnabled to false
- See that OIDC login is disabled

todo:
- [ ] updated docs in case of new feature
- [ ] added/updated tests
- [ ] added/updated testplan to include a test for this fix, including ref to bug using # notation

### CODE REVIEW

**Review by esthervanenckevort — CHANGES_REQUESTED** [review]
This latest change can leave you in a situation where it is very hard to recover from.
I would do it the other way around. isOidcEnabled only functioning as a override to temporarily disable OIDC without removing the configuration. If there is no OIDC configuration in the environment then the isOidcEnabled flag is removed from the settings database.

**Comment by mswertz** [comment]
> This latest change can leave you in a situation where it is very hard to recover from. I would do it the other way around. isOidcEnabled only functioning as a override to temporarily disable OIDC without removing the configuration. If there is no OIDC configuration in the environment then the isOidcEnabled flag is removed from the settings database.

But that would lead to weird dissapearing if a user made the setting. Issen't that confusing?
Harm and I though we should invest in error logging and then in another story make error visible to UI.

File: backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlDatabase.java [hunk 1]
```diff
@@ -179,9 +178,36 @@ public void init() { // setup default stuff
...
+            EnvironmentProperty.getParameter(Constants.MOLGENIS_OIDC_CLIENT_ID, null, STRING);
+        if (envSetting != null) {
+          this.setSetting(Constants.IS_OIDC_ENABLED, "true");
+          this.isOidcEnabled = true;
+        }
+      }
+      // check if OIDC settings are complete otherwise log error and set to false
+      if (this.isOidcEnabled) {
+        if (EnvironmentProperty.getParameter(Constants.MOLGENIS_OIDC_CLIENT_SECRET, null, STRING)
+            == null) {
```
Comment (mswertz) [inline backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlDatabase.java:196]: I think we agreed to not do this (my bad). So all OIDC parameters go via ENV except the OidcEnabled=true/false

**Review by esthervanenckevort — APPROVED** [review]
Tested on localhost and I can enable and disable OIDC.

One thing we might consider is to force log out everyone when OIDC is disabled or enabled.

**Review by mswertz — APPROVED** [review]

**Pull request merged**